### ISSUE TITLE
Support for copying snapshot

### ISSUE THREAD

**Issue by aledeniz** [hashicorp/terraform-provider-azurerm#28210]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

From the Portal it is possible to copy incremental snapshots across regions. 
In the incremental snapshot overview blade the "Copy Snapshot" item is not greyed out. Clicking on that we are redirected to this URI:
https://portal.azure.com/#create/Microsoft.Snapshot-ARM
Once the relevant fields are filled, the creation of a snapshot starts.
This snapshot can be captured in the Terraform state, but the azurerm_snapshot has got a create_option attribute CopyStart which is currently (3.108) not supported. Only Copy and Import are supported. At the moment I am ignoring the create_option, but this means I can only perform this from the Portal/PowerShell/CLI/API, not from Terraform.
Also, the azurerm_snapshot is not immediately performed. The Portal shows a completion_percent attribute, the snapshot is usable to create a new managed disk only after this attribute hits 100.

### New or Affected Resource(s)/Data Source(s)

azurerm_snapshot

### Potential Terraform Configuration

```hcl
..
resource "azurerm_snapshot" "target" {
  name                          = var.target_name
  resource_group_name           = var.target_resource_group.name
  location                      = var.target_location
  create_option                 = "CopyStart" # this is currently missing
  source_uri                    = azurerm_snapshot.source.id
  incremental_enabled           = true # it should work also with false, the requirement for this to be true is for the source
  network_access_policy         = "DenyAll"
  public_network_access_enabled = "false"
  tags                          = var.target_tags
  depends_on                    = [azurerm_snapshot.source]
}
..
output "snapshot_completion_percent" {
  value = azurerm_snapshot.target.completion_percent
}
```

### References

https://learn.microsoft.com/en-us/azure/virtual-machines/disks-copy-incremental-snapshot-across-regions?tabs=azure-cli

https://learn.microsoft.com/en-us/azure/virtual-machines/scripts/copy-snapshot-to-storage-account

**Comment by katbyte** [comment]
Good news, support for `create_option` on `azurerm_snapshot` appears to have shipped in v5.2.0 via #32834, as such we are going to close this request out as completed.

If this does not cover your use case on the latest version of the provider (v5.x) please let us know with a comment or open a new issue referencing this one. Thank you!
